packages/javascript/src/index.ts

```diff
@@ -194,22 +221,38 @@ export default class Appsignal implements JSClient {
     }
   }
 
+  /**
+   * Records and sends a browser `Error` to AppSignal. An alias to `#send()`
+   * to maintain compatibility.
+   */
+  sendError<T>(error: Error, callback?: (span: Span) => T): Promise<Span> | void
+
+  /**
+   * Records and sends a browser `Error` to AppSignal. An alias to `#send()`
+   * to maintain compatibility.
+   */
+  sendError<T>(
+    error: Error,
+    tags?: object,
+    namespace?: string
+  ): Promise<Span> | void
```

Comment:
```suggestion
  sendError<T>(error: Error): Promise<Span> | void
  sendError<T>(error: Error, callback: (span: Span) => T): Promise<Span> | void
  sendError<T>(
    error: Error,
    tags: object,
    namespace: string
  ): Promise<Span> | void
```

I think you can overload the function this way and then make the parameters optional at `public sendError<T>` on line 250

Reply:
👍 I think I'll still need to mark `namespace` as optional. For when the `tags` argument is given as an object, but no `namespace`.

Reply:
I also have to mark tags as optional.

```
src/index.ts:321:31 - error TS2345: Argument of type 'object | ((span: Span) => T) | undefined' is not assignable to parameter of type 'object'.
  Type 'undefined' is not assignable to type 'object'.

321       await this.sendError(e, tagsOrFn, namespace)
                                  ~~~~~~~~

  src/index.ts:238:10
    238   public sendError<T>(
                 ~~~~~~~~~
    The call would have succeeded against this implementation, but implementation signatures of overloads are not externally visible.
```